src/App.tsx

```diff
@@ -15,71 +15,86 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+export class App extends React.Component<{}, {}> {
+  state = {
+    selectedGood: 'Jam',
+  };
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  clearGood = () => {
+    this.setState({ selectedGood: '' });
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  selectBtn = (item: string) => {
+    const { selectedGood } = this.state;
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+    if (selectedGood !== item) {
+      this.setState({ selectedGood: item });
+    } else {
+      this.setState({ selectedGood: '' });
+    }
```

Comment:
should work this way. If it doesn't try creating a variable for the future value of the selected good - and just assign it in the setState
```suggestion
  this.setState({
    selectedGood: selectedGood === item
      ? item
      : ''
  })
```